src/config/mongoose_config_parser_toml.erl

```diff
@@ -306,12 +309,22 @@ get_hosts(Config) ->
         [#config{value = Hosts}] -> Hosts
     end.
 
--spec build_state([jid:server()], [top_level_config()], [override()]) ->
+-spec get_host_types(config_list()) -> [jid:server()].
+get_host_types(Config) ->
+    case lists:filter(fun(#config{key = host_types}) -> true;
+                         (_) -> false
+                      end, Config) of
+        [] -> [];
+        [#config{value = HostTypes}] -> HostTypes
+    end.
```

Comment:
Minor: code duplication, the previous function is very similar. Not mandatory to fix.